src/Services/EncounterService.php

```diff
@@ -17,75 +17,72 @@
 namespace OpenEMR\Services;
 
 use OpenEMR\Common\Acl\AclMain;
+use OpenEMR\Common\Uuid\UuidRegistry;
 use OpenEMR\Services\FacilityService;
+use OpenEMR\Validators\EncounterValidator;
+use OpenEMR\Validators\ProcessingResult;
 use Particle\Validator\Validator;
 
-require_once(dirname(__FILE__) . "/../../library/forms.inc");
-require_once(dirname(__FILE__) . "/../../library/encounter.inc");
+require_once dirname(__FILE__) . "/../../library/forms.inc";
+require_once dirname(__FILE__) . "/../../library/encounter.inc";
 
-
-class EncounterService
+class EncounterService extends BaseService
 {
-
-  /**
-   * Default constructor.
-   */
+    private $encounterValidator;
+    private $uuidRegistery;
+    private const ENCOUNTER_TABLE = "form_encounter";
+    private const PATIENT_TABLE = "patient_data";
+
+    /**
+     * Default constructor.
+     */
     public function __construct()
     {
+        parent::__construct('form_encounter');
+        $this->uuidRegistery = new UuidRegistry(['table_name' => self::ENCOUNTER_TABLE]);
+        $this->uuidRegistery->createMissingUuids();
```

Comment:
Note this createMissingUuids() function is just meant for OpenEMR upgrades. Since uuid is a unique key, mysql will throw an error if there are any empty uuid's (ie. no need to run above line). This is the nice thing about the unique key part of the uuid, it will let us know with a hard error if we have missed any places when inserting form_encounter in the codebase, which can then fix.

Reply:
If I would remove the function and then try to add any Encounter from our UI then the UUID field is empty. And when trying to fetch all records it shows :arrow_down: 
![image](https://user-images.githubusercontent.com/47683556/83397040-9985a900-a41a-11ea-9c44-8a381d91c249.png)
:confused: